src/models/BanList.ts

```diff
@@ -182,6 +182,38 @@ class BanList extends EventEmitter {
         return [...this.serverRules, ...this.userRules, ...this.roomRules];
     }
 
+    /**
+     * Return all of the rules in this list that will match the provided entity.
+     * If the entity is a user, then we match the domain part against server rules too.
+     * @param ruleKind The type of rule for the entity e.g. `RULE_USER`.
+     * @param entity The entity to test e.g. the user id or server name.
```

Comment:
Nit: Likewise, what can we use here exactly?

Reply:
This shouldn't be an exhaustive list, it is supposed to deal with any kind